src/main/java/uk/gov/hmcts/divorce/common/event/page/GeneralApplicationUploadDocument.java

```diff
@@ -38,8 +32,8 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseDa
         final CaseData caseData = details.getData();
         final GeneralApplication generalApplication = caseData.getGeneralApplication();
 
-        if (isNull(generalApplication.getGeneralApplicationDocument())
-            || isNull(generalApplication.getGeneralApplicationDocument().getDocumentLink())) {
+        if (isNull(generalApplication.getGeneralApplicationDocuments())
+            || isEmpty(generalApplication.getGeneralApplicationDocuments())) {
```

Comment:
I would suggest to check the behaviour where a user does not upload any document but fill other fields in the upload document page. A general application document attachment is mandatory when uploading files in general application event.